linked_list.py

```diff
@@ -0,0 +1,77 @@
+class Node(object):
+    def __init__(self, data, next_node=None):
+        self.data, self.next = data, next_node
+
+
+class LinkedList(object):
+    """Create and object-->linked_list."""
+    def __init__(self, head=None):
+        self.head = head
+
+    def insert(self, data):
+        """Put the given value into Node at head of list."""
+        inserted_node = Node(data)
+        inserted_node.next = self.head
+        self.head = inserted_node
+
+    def pop(self):
+        """Return node data, remove node from head of list."""
+        if self.head:
+            value = self.head.data
+            self.head = self.head.next
+            return value
+
+    def size(self):
+        """Return length of linked list."""
+        current_size = self.head
+        list_size = 0
+        while current_size is not None:
+            list_size = list_size + 1
+            current_size = current_size.next
+        print list_size
+        return list_size
+
+    def search(self, data):
+        find_node = self.head
+        found = False
+        while find_node is not None and not found:
```

Comment:
Will you *ever* use the value of `found` if it is `True`?  Check your assumptions.